x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -383,6 +397,14 @@ export function copyAllowlistedFields(
     if (eventValue !== null && eventValue !== undefined) {
       if (allowValue === true) {
         return { ...newEvent, [allowKey]: eventValue };
+      } else if (typeof allowValue === 'object' && Array.isArray(eventValue)) {
+        return {
+          ...newEvent,
+          // cast to object[] to be a valid SearchTypes variant
+          [allowKey]: (eventValue as object[]).map((v) =>
```

Comment:
It seems like this cast isn't safe because even if `eventValue` is an array, it might be an array of strings, boolean, or any other object, according to `SearchTypes`.

Reply:
fair point. yeah i think this assumes that you'll never find something inconsistent with the schema

Reply:
64c9b7a9366ed8dc0337b763e9b576498f3918aa